File: app/src/main/java/com/grarak/ytfetcher/views/recyclerview/settings/ButtonItem.java
package com.grarak.ytfetcher.views.recyclerview.settings;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.grarak.ytfetcher.R;
import com.grarak.ytfetcher.views.recyclerview.RecyclerViewItem;

public class ButtonItem extends RecyclerViewItem<RecyclerView.ViewHolder> {
    private CharSequence text;
    private int textColor;
    private int backgroundColor;
    private View.OnClickListener onClickListener;

    private RecyclerView.ViewHolder viewHolder;

    @Override
    protected int getLayoutXml() {
        return R.layout.item_button;
    }

    @Override
    protected RecyclerView.ViewHolder createViewHolder(View inflatedView) {
        return new RecyclerView.ViewHolder(inflatedView) {
        };
    }

    @Override
    protected void bindViewHolder(RecyclerView.ViewHolder viewHolder) {
        this.viewHolder = viewHolder;

        setup();
    }

    public void setText(CharSequence text) {
        this.text = text;
        setup();
    }

    public void setTextColor(int color) {
        textColor = color;
        setup();
    }

    public void setBackgroundColor(int color) {
        backgroundColor = color;
        setup();
    }

    public void setOnClickListener(View.OnClickListener onClickListener) {
        this.onClickListener = onClickListener;
        setup();
    }

    private void setup() {
        if (viewHolder != null) {
            TextView textView = viewHolder.itemView.findViewById(R.id.text);
            textView.setText(text);
            if (textColor != 0) {
                textView.setTextColor(textColor);
            }
            if (backgroundColor != 0) {
                textView.setBackgroundColor(backgroundColor);
            }

            viewHolder.itemView.setOnClickListener(onClickListener);
        }
    }
}

File: app/src/main/java/com/grarak/ytfetcher/views/recyclerview/settings/ButtonItem.kt
package com.grarak.ytfetcher.views.recyclerview.settings

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView
import com.grarak.ytfetcher.R
import com.grarak.ytfetcher.views.recyclerview.RecyclerViewItem

class ButtonItem(private val onClickListener: View.OnClickListener) : RecyclerViewItem() {
    var text: CharSequence? = null
        set(value) {
            field = value
            setup()
        }
    var textColor = 0
        set(value) {
            field = value
            setup()
        }
    var backgroundColor = 0
        set(value) {
            field = value
            setup()
        }

    private var viewHolder: RecyclerView.ViewHolder? = null

    override val layoutXml = R.layout.item_button

    override fun bindViewHolder(viewHolder: RecyclerView.ViewHolder) {
        this.viewHolder = viewHolder

        setup()
    }

    private fun setup() {
        viewHolder?.run {
            val textView = itemView.findViewById<TextView>(R.id.text)
            textView.text = text
            if (textColor != 0) {
                textView.setTextColor(textColor)
            }
            if (backgroundColor != 0) {
                textView.setBackgroundColor(backgroundColor)
            }

            itemView.setOnClickListener(onClickListener)
        }
    }
}
